Userland/Services/WebDriver/Session.cpp

```diff
@@ -106,4 +114,55 @@ Web::WebDriver::Response Session::stop()
     return JsonValue {};
 }
 
+// 11.4 Get Window Handles, https://w3c.github.io/webdriver/#dfn-get-window-handles
+Web::WebDriver::Response Session::get_window_handles() const
+{
+    // 1. Let handles be a JSON List.
+    JsonArray handles {};
+
+    // 2. For each top-level browsing context in the remote end, push the associated window handle onto handles.
+    for (auto const& window_handle : m_windows.keys()) {
+        handles.append(JsonValue(window_handle));
+    }
+
+    // 3. Return success with data handles.
+    return JsonValue { move(handles) };
+}
+
+Web::WebDriver::Response Session::switch_to_window(StringView const& handle)
+{
+    // 4. If handle is equal to the associated window handle for some top-level browsing context in the
+    //    current session, let context be the that browsing context, and set the current top-level
+    //    browsing context with context.
+    //    Otherwise, return error with error code no such window.
+    auto const& maybe_window = m_windows.get(handle);
+
+    if (!maybe_window.has_value())
+        return Web::WebDriver::Error::from_code(Web::WebDriver::ErrorCode::NoSuchWindow, "Window not found");
+
+    m_current_window_handle = String::from_utf8(handle).release_value_but_fixme_should_propagate_errors();
+
+    // FIXME: 5. Update any implementation-specific state that would result from the user selecting the current
+    //          browsing context for interaction, without altering OS-level focus.
+
+    // 6. Return success with data null.
+    return JsonValue {};
+}
+
+Web::WebDriver::Response Session::close_window()
+{
+    // 1. If the current top-level browsing context is no longer open, return error with error code no such window.
+    // 2. Handle any user prompts and return its value if it is an error.
```

Comment:
This PR also has a bit of a mix of comment spec style when the spec steps are split across the WebDriver and WebContent processes. In `Session::switch_to_window`, you've documented only steps 4-6, as steps 1-3 are implemented in the WebContent process. But here's you've listed steps 1 & 2, which are also implemented in the WebContent process.

IMO we should standardize on the former - only document the spec steps that are actually to be implemented in this function in this process. In order words, steps 1-2 should be removed from here as they're already documented in `WebDriverConnection::close_window`.

Reply:
yeah, agree, I removed steps 1-2. thank you.